CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/resourcetypes/CVKImage.java

```diff
@@ -455,28 +600,353 @@ public static CVKImage Create(  CVKDevice cvkDevice,
+            imageBlitRegion.srcSubresource().aspectMask(VK_IMAGE_ASPECT_COLOR_BIT);
+            imageBlitRegion.srcSubresource().layerCount(1);
+            imageBlitRegion.srcOffsets(1, blitSize);
+            imageBlitRegion.dstSubresource().aspectMask(VK_IMAGE_ASPECT_COLOR_BIT);
+            imageBlitRegion.dstSubresource().layerCount(1);
+            imageBlitRegion.dstOffsets(1, blitSize);
+
+            // Issue the blit command
+            // The blit will automatically transfer from the original format
+            // to the destination. For example the swap image uses RGB 
+            // while the destination format is VK_FORMAT_R8G8B8A8_UNORM 
+            // TODO HYDRA we could probably just use a 3 byte format without alpha
+            // If that works it will save a lot of messing around later. Test after
+            // hittester is working.
+            vkCmdBlitImage(
+                    cvkCopyCmd.GetVKCommandBuffer(),
+                    srcImage.GetImageHandle(), VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL,
+                    destImage.GetImageHandle(), VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL,
+                    imageBlitRegion,
+                    VK_FILTER_NEAREST);
+        }
+        else
+        {
+            // TODO Hydra test this code path
+            // Otherwise use image copy (requires us to manually flip components)
+            VkImageCopy.Buffer imageCopyRegion = VkImageCopy.callocStack(1, stack);
+            imageCopyRegion.srcSubresource().aspectMask(VK_IMAGE_ASPECT_COLOR_BIT);
+            imageCopyRegion.srcSubresource().layerCount(1);
+            imageCopyRegion.dstSubresource().aspectMask(VK_IMAGE_ASPECT_COLOR_BIT);
+            imageCopyRegion.dstSubresource().layerCount(1);
+
+            VkExtent3D extent3D = VkExtent3D.callocStack(stack);
+            extent3D.set(width, height, 1);
+            imageCopyRegion.extent(extent3D);
+
+            // Issue the copy command
+            vkCmdCopyImage(
+                    cvkCopyCmd.GetVKCommandBuffer(),
+                    srcImage.GetImageHandle(), VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL,
+                    destImage.GetImageHandle(), VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL,
+                    imageCopyRegion);
+
+            needsSwizzle = true;
+        }
+
+        // Transition destination image to general layout, which is the required layout for mapping the image memory later on
+        ret = destImage.Transition(cvkCopyCmd, VK_IMAGE_LAYOUT_GENERAL);
+        checkVKret(ret);
+
+        // Transition back the image to original layout after the blit is done
+        ret = srcImage.Transition(cvkCopyCmd, originalLayout);
+        checkVKret(ret);
+
+        // Submit the commands and wait
+        cvkCopyCmd.EndAndSubmit();
+                
+        return destImage;
+    }
+    
+    
+    public PointerBuffer StartMapMemory(MemoryStack stack, CVKImage destImage) {
```

Comment:
Can these both be created internally so StartMapMemory can be called anywhere?  Also should we call it StartReadMemory to match StartWriteMemory in CVKBuffer?

Reply:
Actually they are both wrong now I think of it as it is read or write.  Maybe CVKBuffer should be changed to StartMemoryMap?

Reply:
I've renamed to StartReadMemory  and EndReadMemory, I can look at changing them to be internal in next change when doing hittester code